src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -974,7 +974,7 @@ internal ConstructorInfo[] GetConstructorsInternal(BindingFlags bindingAttr)
         {
             if (ctors == null)
                 return Array.Empty<ConstructorInfo>();
-            ArrayList l = new ArrayList();
+            List<ConstructorBuilder> result = new List<ConstructorBuilder>();
```

Comment:
```suggestion
            List<ConstructorInfo> result = new List<ConstructorInfo>();
```

Reply:
(This is relevant because we're calling ToArray now to produce the resulting array, and it's going to be typed based on the T specified here.  If this produces a ConstructorBuilder[] rather than a ConstructorInfo[], and the consumer tried to then store a ConstructorInfo into the array, it would fail the covariance check.  So to avoid a breaking change, we want the type of the array to be the same as it was before.)

Reply:
100% agree, thanks